tests/abi/tools/revng-ensure-rft-equivalence/Main.cpp

```diff
@@ -51,6 +52,34 @@ static cl::opt<std::string> OutputFilename("o",
                                                           "filename"),
                                            llvm::cl::value_desc("filename"));
 
+static std::optional<std::pair<model::TypePath, model::TypePath>>
+ensureMatchingIDs(const model::Type::Key &Left,
+                  const model::Type::Key &Right,
+                  model::Binary &Model) {
+  auto &[LeftKind, LeftID] = Left;
+  auto &[RightKind, RightID] = Right;
+  revng_assert(LeftKind == RightKind);
+
+  if (LeftID != RightID) {
+    model::TypePath PathToModify = Model.getTypePath(Right);
+    model::Type &ModifiedType = *PathToModify.get();
+
+    // Since the key is getting modified, `batch_insert` makes sure the types
+    // are sorted again at the end of this block.
+    [[maybe_unused]] auto _ = Model.Types().batch_insert();
```

Comment:
Done:
```cpp
    model::TypePath PathToModify = Model.getTypePath(Right);

    // Since the key is getting modified, it's safer to insert the copy of
    // the type anew and then remove the old instance.
    auto [CopyOfTheType, PathToTheCopy] = Model.makeType(*PathToModify.get());
    const model::Type::Key KeyToErase = PathToModify.get()->key();

    // Replace the ID
    CopyOfTheType.ID() = LeftID;

    // Return the "replacement pair", so that the caller is able to gather them
    // and invoke `TupleTree::replaceReferences` on all of them at once.
    model::TypePath ModifiedPath = Model.recordNewType(CopyOfTheType);

    Model.Types().erase(KeyToErase);
    return std::make_pair(PathToModify, ModifiedPath);
```